### ISSUE TITLE
`never_loop` false positive with struct expressions

### ISSUE THREAD

**Issue by andylizi** [rust-lang/rust-clippy#9001]
### Summary

`never_loop` does not pick up `continue` in struct expression fields.

### Lint Name

`never_loop`

### Reproducer

I tried this code:

```rust
struct Foo {
    f: u8
}

fn my_func1() -> Foo {
    loop {
        break Foo {
            f: if 1 + 1 == 3 {
                0
            } else {
                continue;
            }
        }
    }
}
```

[(playground)](https://play.rust-lang.org/?version=nightly&gist=255863740fc6fdd83f84cfa30eddb23a)

I saw this happen:

```
error: this loop never actually loops
  --> src/lib.rs:6:5
   |
6  | /     loop {
7  | |         break Foo {
8  | |             f: if 1 + 1 == 3 { 0 } else { continue; }
9  | |         }
10 | |     }
   | |_____^
   |
   = note: `#[deny(clippy::never_loop)]` on by default
```

I expected no error, since it will loop forever.

### Version

```text
rustc 1.63.0-nightly (ca122c7eb 2022-06-13)
binary: rustc
commit-hash: ca122c7ebb3ab50149c9d3d24ddb59c252b32272
commit-date: 2022-06-13
host: x86_64-pc-windows-msvc
release: 1.63.0-nightly
LLVM version: 14.0.5
```
